File: app/src/main/java/com/suhang/networkmvp/model/INetworkModel.java
package com.suhang.networkmvp.model;


import com.suhang.networkmvp.presenter.INetworkPresenter;

import java.io.File;
import java.util.Map;

/**
 * Created by sh on 2016/10/25 16:44.
 */

public interface INetworkModel {

	/**
	 * 获取数据的方法
	 *
	 * @param params   post请求需要的参数键值对
	 * @param listener 回调接口
	 */
	<T> void loadData(Class<T> aClass, Map<String, String> params, boolean needCache, int tag, INetworkPresenter.OnDataLoadingListener<T> listener);

	/**
	 * 上传文件
	 * @param params
	 * @param file
	 * @param tag 标识
	 * @param listener
	 */
	<T> void upload(Class<T> aClass, Map<String, String> params, File file, int tag, INetworkPresenter.OnDataLoadingListener<T> listener);

	/**
	 * 下载文件
	 *
	 * @param url
	 * @param listener
	 * @param handler
	 */
	void download(String url, String name, String path, int tag, INetworkPresenter.OnDataLoadingListener listener);

	/**
	 * 取消某个请求或者停止上传
	 *
	 * @param tag 该请求的标记
	 */
	void cancelNormal(int tag);

	/**
	 * 停止下载
	 *
	 * @param tag 该请求的标记
	 */
	void cancelDownload(int tag);

	/**
	 * 取消所有请求,用于防止内存泄漏
	 */
	void cancelAll();
}

File: app/src/main/java/com/suhang/networkmvp/mvp/model/INetworkModel.kt
package com.suhang.networkmvp.mvp.model

import com.suhang.networkmvp.constants.DEFAULT_TAG


/**
 * Created by 苏杭 on 2017/5/23 15:56.
 */
interface INetworkModel {
    companion object{
        val GET: Int = 1
        val POST: Int = 2
    }
    /**
     * 访问网络获取数据(POST)(获取包裹类,并转换为所需bean类)
     * @param aClass Bean类字节码
     *
     * @param append Bean类中URL字段后的附加字段(类似URL1,URL2),用于处理一个Bean类对应多个接口
     *
     * @param needCache 是否需要缓存(默认根据URL缓存)
     *
     * @param cacheTag 缓存附加标志(以URL+cacheTag缓存),用于处理同一个URL,根据传入参数不同得到的数据不同时的缓存方案(POST请求)
     *
     * @param whichTag 标记,用于一个页面同时处理多个获取数据的请求时分辨是哪一个请求
     *
     * @param objects 接口参数(根据Retrofit的Service中对应的方法参数传入)
     */
    fun loadData(requestWay:Int,dataClass: Class<Any>, append: String = "", needCache: Boolean = false, cacheTag: String, whichTag: Int = DEFAULT_TAG,isWrap:Boolean, vararg params: Any)
}